app/src/main/java/io/github/zadam/triliumsender/LoginActivity.kt

```diff
@@ -1,29 +1,28 @@
 package io.github.zadam.triliumsender
 
-import android.os.AsyncTask
 import android.os.Bundle
-import android.support.v7.app.AppCompatActivity
 import android.text.TextUtils
 import android.util.Log
-import android.view.View
 import android.view.inputmethod.EditorInfo
 import android.widget.TextView
 import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
 import io.github.zadam.triliumsender.services.TriliumSettings
 import io.github.zadam.triliumsender.services.Utils
 import kotlinx.android.synthetic.main.activity_login.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
 import okhttp3.OkHttpClient
 import okhttp3.Request
-import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 import org.json.JSONObject
 
 
 class LoginActivity : AppCompatActivity() {
-    /**
-     * Keep track of the login task to ensure we can cancel it if requested.
```

Comment:
Since there wasn't actually an implemented way to cancel the login task, I decided to get rid of this for now. I think a more clear thing to do would be to disable the login button - maybe show a spinner?- while the login coroutine is running, and either hide the spinner / reenable the button on failure, or do what we do now on success- `finish()` the activity.

Reply:
(the worst case scenario here is the user double taps and logs in twice, getting an extra long toast. happy to do the disable+spinner idea in this PR if you're concerned about it, or in a follow-on PR)